kedro/io/data_catalog.py

```diff
@@ -12,6 +12,7 @@
 import re
 from typing import Any, Dict
 
+from omegaconf import OmegaConf
```

Comment:
I don't know how I feel about coupling the `DataCatalog` class with OmegaConf here, let alone special-casing parameters by using the `:` separator.

Shouldn't we redesign the `DataCatalog` API instead so that parameters are first class citizens, and not fake datasets?

Reply:
+1 to that 👆🏻

Reply:
> Would you really call this coupling? The way I read it is that is uses omegaconf to parse the parameters config. We already have a dependency on omegaconf anyway, and I actually quite like that we can leverage it in more places than just the `OmegaConfigLoader` itself. I would have called it coupling if it uses the actual `OmegaConfigLoader` _class_, but this just imports the library.

_Originally posted by @merelcht in https://github.com/kedro-org/kedro/issues/3893#issuecomment-2203434513_